### PR TITLE
Fix #169 -- add systemd env options

### LINKED ISSUE #169: Request for Clarification [setting proxy information]
According to the advanced configuration documentation: https://docs.rke2.io/advanced proxy configurations should be set in `/etc/default/rke2-[server,agent]`.

Is that supported using either settings in `group_vars/...` or `hosts.ini`?

### PR DESCRIPTION
**Pull request by ghost**
## What type of PR is this?

- [ ] bug
- [ ] cleanup
- [ ] documentation
- [ X] feature

## What this PR does / why we need it:
It adds the optional ability to add and configure an rke-{server,agent} systemd environment file to /etc/default

## Which issue(s) this PR fixes:

Fixes #169 

## Release Notes

```release-note
Add the ability to optionally create an /etc/default/rke-{server,agent} file to configure the systemd file for proxy information.
```

### CODE REVIEW

File: inventory/sample/group_vars/rke2_servers.yml [hunk 1]
```diff
@@ -45,3 +45,10 @@ rke2_config: {}
 # See https://docs.rke2.io/helm/#automatically-deploying-manifests-and-helm-charts
 # Add manifest files by specifying the directory path on the control host
 # manifest_config_file_path: "{{ playbook_dir }}/sample_files/manifest/"
+
+# See https://docs.rke2.io/advanced#configuring-an-http-proxy
+# Add proxy information for the systemd environment
+# systemd-extra-env:
```
Comment (Falltrades) [inline inventory/sample/group_vars/rke2_servers.yml:51]:
```suggestion
# systemd_extra_env:
```

File: roles/rke2_common/tasks/add-systemd-env.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+- name: Systemd env file
+  become: true
+  when: (systemd_extra_env is defined) and (systemd_extra_env|length > 0)
+  block:
+
+    - name: Add the systemd env file
+      ansible.builtin.blockinfile:
+        path: /etc/default/rke2-{{ caller_role_name }}
```
Comment (Falltrades) [inline roles/rke2_common/tasks/add-systemd-env.yml:9]:
```suggestion
        path: /etc/default/rke2-{{ rke2_common_caller_role_name }}
```

File: inventory/sample/group_vars/rke2_agents.yml [hunk 3]
```diff
@@ -8,3 +8,10 @@ rke2_config: {}
 # See https://docs.rke2.io/install/containerd_registry_configuration/
 # Add a registry configuration file by specifying the file path on the control host
 # registry_config_file_path: "{{ playbook_dir }}/sample_files/registries.yaml"
+
+# See https://docs.rke2.io/advanced#configuring-an-http-proxy
+# Add proxy information for the systemd environment
+# systemd-extra-env:
```
Comment (Falltrades) [inline inventory/sample/group_vars/rke2_agents.yml:14]:
```suggestion
# systemd_extra_env:
```

File: roles/rke2_common/tasks/add-systemd-env.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        marker: "#{mark} ANSIBLE MANAGED FILE"
+        create: true
+        mode: '640'
+        owner: root
+        group: root
+        block: |
+          {% for item in systemd_extra_env %}
+          {{ item }}
+          {% endfor %}
+    - name: Reload the systemd daemon
```
Comment (Falltrades) [inline roles/rke2_common/tasks/add-systemd-env.yml:19]: Maybe is it better to use handlers ?

**Comment by aleiner** [comment]
address via #242 due to rework

**Pull request closed without merging**